test/unit/helpers/ansible_reports_helper_test.rb

```diff
@@ -19,4 +19,18 @@ class AnsibleReportsHelperTest < ActiveSupport::TestCase
       ansible_module_message(log).to_s
     )
   end
+
+  test 'module message extraction with error' do
+    log_value = <<-ANSIBLELOG.strip_heredoc
+    {"msg": "AnsibleUndefinedVariable", "changed": false, "_ansible_no_log": false, "failed": true, "module": "template", "exception": "raise AnsibleUndefinedVariable"}
+ANSIBLELOG
+    message = FactoryBot.build(:message)
+    message.value = log_value
+    log = FactoryBot.build(:log)
```

Comment:
```ruby
log = FactoryBot.build(:log, message: message)
```

Reply:
This won't work, because of this [line](https://github.com/theforeman/foreman/blob/f3881ee0fca2b8b3d78467a5013c7e4b0347828c/test/factories/reports_related.rb#L41) in Foreman core.
We can change it to: `log.message = FactoryBot.create(:message) unless log.message`, for it to work. The question is if we want to do it.